src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -1093,50 +1095,40 @@ SPECTRE_ALWAYS_INLINE constexpr auto create_from(Box&& box,
                 "Must pass in as many arguments as AddTags to db::create_from");
 
   // 1. Full list of old tags, and the derived tags list of the RemoveTags
-  using old_box_tags = typename std::decay_t<Box>::tags_list;
+  using old_tags = typename std::decay_t<Box>::tags_list;
   static_assert(
-      tmpl::all<RemoveTags, has_unique_matching_tag<tmpl::pin<old_box_tags>,
+      tmpl::all<RemoveTags, has_unique_matching_tag<tmpl::pin<old_tags>,
                                                     tmpl::_1>>::value,
       "One of the tags being removed could not be found in the DataBox or "
       "is a base tag identifying more than one tag.");
-  using remove_tags = tmpl::transform<
-      RemoveTags,
-      tmpl::bind<first_matching_tag, tmpl::pin<old_box_tags>, tmpl::_1>>;
-
-  // 2. Expand simple remove tags and compute remove tags
-  using compute_tags_to_remove =
-      tmpl::filter<remove_tags, db::is_compute_tag<tmpl::_1>>;
-  using compute_tags_to_remove_with_subitems =
-      expand_subitems<compute_tags_to_remove>;
-  using simple_tags_to_remove =
-      tmpl::list_difference<remove_tags, compute_tags_to_remove>;
-  using simple_tags_to_remove_with_subitems =
-      expand_subitems<simple_tags_to_remove>;
-
-  // 3. Expand AddTags (these are just the simple tags)
-  using simple_tags_to_add_with_subitems = expand_subitems<AddTags>;
-  using compute_tags_to_add_with_subitems = expand_subitems<AddComputeTags>;
-
-  // 4. Create new list of tags by removing all the remove tags, and adding all
-  // the AddTags, including subitems
-  using simple_tags_to_keep =
+  using remove_tags =
+      tmpl::transform<RemoveTags, tmpl::bind<first_matching_tag,
+                                             tmpl::pin<old_tags>, tmpl::_1>>;
+
+  // 2. Expand subitems of tags to remove
+  using immutable_item_tags_to_remove = expand_subitems<
+      tmpl::filter<remove_tags, db::is_immutable_item_tag<tmpl::_1>>>;
+  using mutable_item_tags_to_remove = expand_subitems<
+      tmpl::list_difference<remove_tags, immutable_item_tags_to_remove>>;
```

Comment:
Can this be done with a filter with `tmpl::not`? I haven't run any tests on this, but reasoning through the tmp operations, it seems that filtering with an easily compiled metafunction should be faster than `list_difference` if it gives the same result.